hack/changelog-gen.sh

```diff
@@ -49,31 +51,31 @@ if [ "$(git rev-parse --abbrev-ref HEAD)" == "master" ]; then
   exit 1
 fi
 
-if git tag -l|grep -q $NEW_RELEASE; then
+if git tag -l | grep -q $NEW_RELEASE; then
   echo "Error: Tag $NEW_RELEASE already exists!"
   exit 1
 fi
 
-if ! echo -n $NEW_RELEASE|egrep -q '^v(0|[1-9]\d*)\.(0|[1-9]\d*)\.(0|[1-9]\d*)$'; then
+if ! echo -n $NEW_RELEASE | egrep -q '^v(0|[1-9]\d*)\.(0|[1-9]\d*)\.(0|[1-9]\d*)$'; then
   echo "Error: Release version \"$NEW_RELEASE\" does not match regex '^v(0|[1-9]\d*)\.(0|[1-9]\d*)\.(0|[1-9]\d*)$'"
   echo "Valid version must look like this: v1.2.3"
   exit 1
 fi
 
 # Install gchl if not installed
 if ! [ -x "$(command -v gchl)" ]; then
-	echo "gchl not installed!"
-	echo "Executing go get on github.com/kubermatic/gchl ..."
-	go get github.com/kubermatic/gchl
-	echo "Done!"
+  echo "gchl not installed!"
+  echo "Executing go get on github.com/kubermatic/gchl ..."
+  go get github.com/kubermatic/gchl
+  echo "Done!"
 fi
 
 # Create CHANGELOG.md if not exists in order for the cat not to fail
-[ -f $CHANGELOG_FILE ] ||  touch $CHANGELOG_FILE
+[ -f $CHANGELOG_FILE ] || touch $CHANGELOG_FILE
 
 # Generate changelog
 OUTPUT="$(gchl --for-version $NEW_RELEASE since $(git describe --abbrev=0 --tags) --release-notes)"
-echo "${OUTPUT}" | cat - $CHANGELOG_FILE > temp
+echo "${OUTPUT}" | cat - $CHANGELOG_FILE >temp
```

Comment:
Why does shfmt do The Right Thing by putting spaces around pipes, but does The Bad Thing by not having spaces for output redirects? Why is `a | b` right, but `a > b` is wrong and needs to be `a >b`?

Reply:
https://github.com/mvdan/sh/issues/243#issuecomment-392019995

Reply:
Meh, I am not really convinced, but this is not a hill I want to die on. You can have your redirects, as long as we never ever indent shell scripts with tabs.

Reply:
There is now a flag (`-sr`) specified both prow and Makefile